ownCloud/Bookmarks/BookmarkManager.swift

```diff
@@ -113,6 +113,26 @@ class BookmarkManager: NSObject {
 
 		return bookmark
 	}
+
+    func removeAuthDataOfBookmark(_ bookmark: OCBookmark) -> OCBookmark {
+        
+        if bookmark.authenticationMethodIdentifier == OCAuthenticationMethodBasicAuthIdentifier {
+            let username = OCAuthenticationMethodBasicAuth.userName(fromAuthenticationData: bookmark.authenticationData)
+            let password = ""
+            
+            do {
+                try bookmark.authenticationData = OCAuthenticationMethodBasicAuth.authenticationData(forUsername: username, passphrase: password, authenticationHeaderValue: nil)
```

Comment:
Don't use this. From the docs / comment for this method:

> Generates authentication data for basic auth without reaching out to the server. Useful for building tests. **Should not be used for anything other than implementing tests.**

Reply:
Ok I used it because do exactly what I need. "Generates authentication data for basic auth without reaching out to the server".
In that point I am deleting the password for a Basic credentials but I want to still storing the username.

I will use then the (ignoring errors):
```
- (void)generateAuthenticationDataWithMethod:(OCAuthenticationMethodIdentifier)methodIdentifier options:(OCAuthenticationMethodBookmarkAuthenticationDataGenerationOptions)options completionHandler:(void(^)(NSError *error, OCAuthenticationMethodIdentifier authenticationMethodIdentifier, NSData *authenticationData))completionHandler;
```

Reply:
The behavior of the Edit was implemented as you say. It just a problem on the behavior of the button "Delete Authentication Data".

I will remove it on the case of Basic Auth.

Reply:
> A "Delete bookmark" button instead the "Delete Authentication Data" it is not necessary because we have a Delete button in the top right.
> 
> <img width="246" alt="screen shot 2018-04-20 at 10 30 58" src="https://user-images.githubusercontent.com/1815050/39039543-f7b130b8-4485-11e8-8bf8-ab27caf7564c.png">

Right 😅

Reply:
Ops, ok. I will change the presentation 👌